src/graph.cpp

```diff
@@ -44,8 +32,11 @@ std::shared_ptr<Vertex> Graph::findVertex(const unsigned int &id) const {
  * @return True if successful, and false if a vertex with the given id already exists
  */
 std::shared_ptr<Vertex> Graph::addVertex(const unsigned int &id, Coordinates c) {
-    std::shared_ptr<Vertex> newVertex = std::make_shared<Vertex>(id, c);
-    vertexSet.insert(newVertex);
+    std::shared_ptr<Vertex> newVertex = nullptr;
+    if (vertexSet.size() <= id) {
```

Comment:
If I have an Vertex to add with the id 12, but the vertexSet has size 9, the 12 vertex will be added to the 10th slot and break the findVertex logic. Are we just trusting that vertices are added in order?